subprojects/docs/src/docs/userguide/repositoryTypes.adoc

```diff
@@ -0,0 +1,390 @@
+
+| `file`
+| none
+
+| `http`
+| username/password
+
+| `https`
+| username/password
+
+| `sftp`
+| username/password
+
+| `s3`
+| access key/secret key/session token or Environment variables
+
+| `gcs`
+| https://developers.google.com/identity/protocols/application-default-credentials[default application credentials] sourced from well known files, Environment variables etc.
+|===
+
+[NOTE]
+====
+Username and password should never be checked in plain text into version control as part of your build file. You can store the credentials in a local `gradle.properties` file and use one of the open source Gradle plugins for encrypting and consuming credentials e.g. the link:https://plugins.gradle.org/plugin/nu.studer.credentials[credentials plugin].
+====
+
+The transport protocol is part of the URL definition for a repository. The following build script demonstrates how to create a HTTP-based Maven and Ivy repository:
+
+++++
+<sample id="mavenIvyRepositoriesNoAuth" dir="userguide/artifacts/defineRepository" title="Declaring a Maven and Ivy repository">
+    <sourcefile file="build.gradle" snippet="maven-ivy-repository-no-auth"/>
+</sample>
+++++
+
+The following example shows how to declare SFTP repositories:
+
+++++
+<sample id="mavenIvyRepositoriesAuth" dir="userguide/artifacts/defineRepository" title="Using the SFTP protocol for a repository">
+    <sourcefile file="build.gradle" snippet="maven-ivy-repository-auth"/>
+</sample>
+++++
+
+When using an AWS S3 backed repository you need to authenticate using api:org.gradle.api.credentials.AwsCredentials[], providing access-key and a private-key. The following example shows how to declare a S3 backed repository and providing AWS credentials:
+
+++++
+<sample id="mavenIvyS3RepositoriesAuth" dir="userguide/artifacts/defineRepository" title="Declaring a S3 backed Maven and Ivy repository">
+    <sourcefile file="build.gradle" snippet="maven-ivy-s3-repository"/>
+</sample>
+++++
+
+You can also delegate all credentials to the AWS sdk by using the AwsImAuthentication. The following example shows how:
+
+++++
+<sample id="mavenIvyS3RepositoriesAuthWithIam" dir="userguide/artifacts/defineRepository" title="Declaring a S3 backed Maven and Ivy repository using IAM">
+    <sourcefile file="build.gradle" snippet="maven-ivy-s3-repository-with-iam"/>
+</sample>
+++++
+
+When using a Google Cloud Storage backed repository default application credentials will be used with no further configuration required:
+
+++++
+<sample id="mavenIvyGCSRepositoriesAuthDefault" dir="userguide/artifacts/defineRepository" title="Declaring a Google Cloud Storage backed Maven and Ivy repository using default application credentials">
+    <sourcefile file="build.gradle" snippet="maven-ivy-gcs-repository"/>
+</sample>
+++++
+
+[[sub:s3_configuration_properties]]
+==== S3 configuration properties
+
+The following system properties can be used to configure the interactions with s3 repositories:
+
+.S3 Configuration Properties
```

Comment:
Should be "S3 configuration properties" for sentence case

Reply:
done